tutorials/super_blink/research/adc.c

```diff
@@ -0,0 +1,87 @@
+#include "libs/adc/api.h"
+
+#include <avr/interrupt.h>
+#include <avr/io.h>
+#include <stdlib.h>
+
+void (*interrupt_callback)(void) = NULL;
+
+ISR(ADC_vect) {
+    if (interrupt_callback != NULL) {
+        (*interrupt_callback)();
+    }
+}
+
+/*
+ * Initialize the adc on the MCU
+ *
+ * ADCSRA = ADC register A containing configuration bits (ADC Control and Status
+ * Register A)
+ *
+ * ADCSRB = ADC register B containing more configuration bits
+ *
+ * ADEN = ADC enable bit
+ *
+ * ADPS2:0 = ADC prescaler select - division factor between
+ * system clock fq and input clock to ADC
+ *
+ * AREFEN = Analog reference pin enable
+ *
+ * ADMUX = Register for muxing to various ADC inputs
+ *
+ * REFS0 = REFSn: Selects reference for the ADC
+ */
+
+// Initializes the ADC on the MCU
+void adc_init(void) {
+    ADCSRA = (1 << ADEN) | (1 << ADPS2)
+             | (1 << ADPS0); // Sets the ADEN, ADPS2, and ADPS0 bits to 1.
+                             // Enables the ADC, sets the prescaler to 32
+    ADCSRB &= (1 << AREFEN); // What does this do? Clear all bits but AREFEN?
```

Comment:
What is the purpose of this line? Clear all bits but `AREFEN`? Initial values for the bits in `ADCSRB` should be all 0, so clearing all bits but `AREFEN` (which is also 0) should set `ADCSRB` to `0x00`, no?